### PR TITLE
Replace Run with RunE

### LINKED ISSUE #479: [nextercism] cobra.Command: Run vs RunE
Cobra has a RunE function that returns an error, which propagates to Execute, which we then can detect and use to exit just the once: https://github.com/exercism/cli/blob/nextercism/cmd/root.go#L29-L34

I think this would be a vast improvement over the hacky BailOnError thing we have going on.

### PR DESCRIPTION
**Pull request by robphoenix**
Closes #479

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -17,7 +17,10 @@ var (
 	// The usage examples and help strings should reflect
 	// the actual name of the binary.
 	BinaryName string
-	Out        io.Writer
+	// Out is used by the download command
```
Comment (nywilken) [inline cmd/root.go]: `Out` should actually be used in multiple commands. It looks like we have `out` and `Out` defined in the cmd package. If #483 is good by you, we should merge those changes in then rebase your branch onto master.

**Comment by robphoenix** [comment]
Hope you don't mind, I went ahead and merged #483, and have rebased this PR.

**Comment by nywilken** [comment]
> Hope you don't mind, I went ahead and merged #483, and have rebased this PR.

Not at all, thanks.

**Comment by nywilken** [comment]
I pulled down your latest changes for some final testing and found a quirk with the error output. More specifically, when the command exits with error it prints the error followed by the usage text. 

```Bash
› ./fakeexercism version --latest
exercism version 3.0.0-alpha.1
Error: Get https://api.github.com/repos/exercism/cli/releases/latest: dial tcp: lookup api.github.com: no such host
Usage:
   version [flags]

Aliases:
  version, v

Flags:
  -h, --help     help for version
  -l, --latest   check latest available version

Get https://api.github.com/repos/exercism/cli/releases/latest: dial tcp: lookup api.github.com: no such host
```

Looking at the documentation for [Cobra](https://github.com/spf13/cobra#alternative-error-handling) I see that there are two flags `SilenceUsage` and `SilenceErrors` flags that can be set in the `RootCmd` to disable the display of the usage text on error. 

```Go
 24 // RootCmd represents the base command when called without any subcommands.
 25 var RootCmd = &cobra.Command{
 26   Use:   BinaryName,
 27   Short: "A friendly command-line interface to Exercism.",
 28   Long: `A command-line interface for https://v2.exercism.io.
 29
 30 Download exercises and submit your solutions.`,
 31   SilenceUsage: true,
 32   SilenceErrors: true,
 33 }
```

Which results in the following output

```Bash
› ./fakeexercism version --latest
exercism version 3.0.0-alpha.1
Get https://api.github.com/repos/exercism/cli/releases/latest: dial tcp: lookup api.github.com: no such host
```

If you want to use the default error messaging format for RunE you can drop the call to fmt.Println at [line 36](https://github.com/exercism/cli/pull/482/files#diff-ff7686b39bf90dc2520886fb874371a4R36) and drop the drop the flag `SilenceErrors`

**Comment by robphoenix** [comment]
> If you want to use the default error messaging format for RunE you can drop the call to fmt.Println at line 36 and drop the flag SilenceErrors

Excellent, good catch @nywilken, thanks!

**Comment by kytrinyx** [comment]
This is so much nicer. Thank you for running with it!

**Comment by nywilken** [comment]
@robphoenix I went ahead and rebased (just in case) & merged the PR. I left it sitting for a while, not knowing that you actually added the SilenceUsage flag 🤦‍♂️

Great work!

**Comment by kytrinyx** [comment]
Thank you both so much for keeping things moving forward with the CLI in the past three or so weeks. I've had one of those "everything is too much" crises and dropped all of the balls. All of them. Especially the CLI ones. I'm slowly working my way back into things and feeling a lot better and it's _ever_ so much nicer to come back when someone has kept the lights on. It feels so much less lonely and insurmountable.

So anyway. Thank you.

✨ 💛 ✨ 🌻 ✨ 💛 ✨

**Comment by robphoenix** [comment]
> So anyway. Thank you.

You're more than welcome. Very happy to help keep the lights on, and assuage any loneliness, and sending many :hugs: and much :heart: your way.

**Pull request merged**